deflect/ServerWorker.cpp

```diff
@@ -268,7 +258,18 @@ void ServerWorker::_handleMessage(const MessageHeader& messageHeader,
         {
             const bool exclusive =
                 (messageHeader.type == MESSAGE_TYPE_BIND_EVENTS_EX);
-            emit registerToEvents(_streamId, exclusive, this);
+            auto promise = std::make_shared<std::promise<bool>>();
+            auto future = promise->get_future();
+            emit registerToEvents(_streamId, exclusive, this,
+                                  std::move(promise));
+            try
+            {
+                _registeredToEvents = future.get();
+            }
+            catch (...)
```

Comment:
I find it safer to catch everything in case the promise holds an exception instead of a value. In all cases, it is reasonable to assume that if an exception happens then the registration must have failed.